addons/binding/org.openhab.binding.smappee/src/main/java/org/openhab/binding/smappee/internal/SmappeeService.java

```diff
@@ -0,0 +1,477 @@
+                                + "&maxNumber=1&from=" + nowUtcMinus1YearMillis + "&to=" + nowUtcMillis);
+
+                if (responseReadings.isEmpty()) {
+                    return null;
+                }
+
+                SmappeeApplianceEvent[] readings = gson.fromJson(responseReadings, SmappeeApplianceEvent[].class);
+
+                logger.debug("smappee '{}' - appliance '{}' read", this.serviceLocationId, applianceId);
+
+                if (readings.length == 1) {
+                    return readings[0];
+                }
+                return null;
+
+            } catch (CommunicationException se) {
+                logger.warn("failed to read smappee '{}' - appliance '{}' : {}", this.serviceLocationId, applianceId,
+                        se.getMessage());
+            } catch (JsonSyntaxException pe) {
+                logger.warn("failed to read response from smappee '{}' - appliance '{}' : {}", this.serviceLocationId,
+                        applianceId, pe.getMessage());
+            }
+        } while (currentTry++ < this.retry);
+
+        return null;
+    }
+
+    /**
+     * Gets the smappee energy consumption readings for a specific sensor.
+     *
+     * @return the device readings
+     */
+    public SmappeeSensorConsumptionRecord getLatestSensorConsumption(String sensorId) {
+        int currentTry = 0;
+        do {
+            try {
+                DateTime nowUtc = DateTime.now(DateTimeZone.UTC).minusMinutes(1);
+                DateTime nowUtcMinus1Year = DateTime.now(DateTimeZone.UTC).minusYears(1);
+
+                String nowUtcMillis = String.valueOf(nowUtc.getMillis());
+                String nowUtcMinus1YearMillis = String.valueOf(nowUtcMinus1Year.getMillis());
+
+                // sample API method to call :
+                // https://app1pub.smappee.net/dev/v1/servicelocation/1/sensor/4/consumption?
+                // from=1457597400000&to=1458666049000&aggregation=1
+                String responseReadings = getData(
+                        "/dev/v1/servicelocation/" + this.serviceLocationId + "/sensor" + sensorId + "/consumption?"
+                                + "aggregation=1&from=" + nowUtcMinus1YearMillis + "&to=" + nowUtcMillis);
+
+                if (responseReadings.isEmpty()) {
+                    return null;
+                }
+
+                SmappeeSensorConsumption readings = gson.fromJson(responseReadings, SmappeeSensorConsumption.class);
+
+                logger.debug("smappee '{}' - sensor '{}' read", this.serviceLocationId, sensorId);
+
+                if (readings.records != null && readings.records.length > 0) {
+                    return readings.records[0];
+                }
+                return null;
+
+            } catch (CommunicationException ce) {
+                logger.warn("failed to read smappee '{}' - sensorId '{}' : {}, Retry ({}/{})", this.serviceLocationId,
```

Comment:
change to debug (and have the status set to COMMUNICATION-ERROR?)

Reply:
OK